tests/unit/test_iceberg_location.py

```diff
@@ -0,0 +1,79 @@
+import pytest
+from dbt.adapters.snowflake.relation import SnowflakeRelation
+
+
+@pytest.fixture
+def iceberg_config() -> dict:
+    """Fixture providing standard Iceberg configuration."""
+    return {
+        "schema": "my_schema",
+        "identifier": "my_table",
+        "external_volume": "s3_iceberg_snow",
+        "base_location_root": "root_path",
+        "base_location_subpath": "subpath",
+    }
+
+
+def get_actual_base_location(config: dict[str, str]) -> str:
+    """Get the actual base location from the configuration by parsing the DDL predicates."""
+
+    relation = SnowflakeRelation.create(
+        schema=config["schema"],
+        identifier=config["identifier"],
+    )
+
+    actual_ddl_predicates = relation.get_iceberg_ddl_options(config).strip()
+    actual_base_location = actual_ddl_predicates.split("base_location = ")[1]
+
+    return actual_base_location
+
+
+def test_iceberg_path_and_subpath(iceberg_config: dict[str, str]):
```

Comment:
This is not needed for merging, but have you thought about turning these four tests into a parameterized tests and making the expected location a string literal? The parameterization would look something like:

```python
@pytest.mark.parameterize("omitted_config,expected_path", [
    ([], "root_path/my_schema/my_table/subpath"),
    (["base_location_root"], "_dbt/my_schema/my_table/subpath"),
    (["base_location_subpath"], "root_path/my_schema/my_table"),
    (["base_location_root", "base_location_subpath"], "_dbt/my_schema/my_table"),
])
```
I could also see it the other way around, where the config gets built dynamically based on what the user provided in each scenario (instead of omitted).